pkg/controllers/openshift_apiserver_monitor/setup.go

```diff
@@ -0,0 +1,55 @@
+package openshift_apiserver_monitor
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/openshift/ibm-roks-toolkit/pkg/cmd/cpoperator"
+	"github.com/openshift/ibm-roks-toolkit/pkg/controllers"
+)
+
+func Setup(cfg *cpoperator.ControlPlaneOperatorConfig) error {
+	apiextClient, err := apiextensionsclient.NewForConfig(cfg.TargetConfig())
+	if err != nil {
+		return err
+	}
+	crdInformer := cache.NewSharedIndexInformer(
+		&cache.ListWatch{
+			ListFunc: func(opt metav1.ListOptions) (runtime.Object, error) {
```

Comment:
Hey! This looks amazing I was just curious on one thing if you can spark my memory:

Just to ensure there's not a load spike in the central control plane it looks like all the lists and watches target the *user* cluster. And then just simple patch calls are made to the *management* cluster but no watches or lists are made (heavy calls that will trigger more load) correct?

Reply:
correct, the watch is on CRDs of the target cluster. On the parent cluster we just read and update the deployment resource.